test/rssfeed.cpp

```diff
@@ -261,23 +261,33 @@ TEST_CASE("RssFeed::get_firsttag() returns first tag", "[RssFeed]")
 	Cache rsscache(":memory:", &cfg);
 	RssFeed f(&rsscache);
 
-	REQUIRE(f.get_firsttag() == "");
+	SECTION("Empty tag array") {
+		REQUIRE(f.get_firsttag() == "");
+	}
 
-	std::vector<std::string> tags = {"One", "Two", "Three", "Four"};
-	f.set_tags(tags);
-	REQUIRE(f.get_firsttag() == "One");
+	SECTION("Ordinary tags") {
+		std::vector<std::string> tags = {"One", "Two", "Three", "Four"};
+		f.set_tags(tags);
+		REQUIRE(f.get_firsttag() == tags[0]);
```

Comment:
Whenever possible, use literal values in assertions, because:

1. when an assertion fails, the values are printed out. Seeing both the result and the expected value makes it quicker to debug, since you don't have to look up the test and find out what `tags[0]` contains;
2. tests in general should be as "obvious" as possible, even at the expense of some repetition.

Also, `tags` are never modified, so either mark it `const` or drop it altogether (you can construct a vector right in the call to `set_tags`: `set_tags({"one", "two", "three"})`. Marking everything `const` makes it harder to commit certain errors, and also documents your intent better.

(This applies to all the other sections, below.)

Reply:
Got it, new commit uploaded.